examples/petset/mongodb-petset.yaml

```diff
@@ -0,0 +1,146 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: mongodb-petset-replica
+  annotations:
+    description: "MongoDB Replication Example (based on PetSet)"
+    iconClass: "icon-mongodb"
+    tags: "database,mongodb,replication"
+parameters:
+  - name: MONGODB_USER
+    displayName: "MongoDB Connection Username"
+    description: "Username for MongoDB user that will be used for accessing the database"
+    generate: expression
+    from: "[a-zA-Z0-9]{3}"
+    required: true
+
+  - name: MONGODB_PASSWORD
+    displayName: "MongoDB Connection Password"
+    description: "Password for the MongoDB connection user"
+    generate: expression
+    from: "[a-zA-Z0-9]{12}"
+    required: true
+
+  - name: MONGODB_DATABASE
+    displayName: "MongoDB Database Name"
+    description: "Name of the MongoDB database accessed"
+    value: sampledb
+    required: true
+
+  - name: MONGODB_ADMIN_PASSWORD
+    displayName: "MongoDB Admin Password"
+    description: "Password for the database admin user"
+    generate: expression
+    from: "[a-zA-Z0-9]{12}"
+    required: true
+
+  - name: MONGODB_REPLICA_NAME
+    displayName: "Replica Set Name"
+    description: "The name of the replica set"
+    value: rs0
+    required: true
+
+  - name: MONGODB_KEYFILE_VALUE
+    displayName: "Key File Content"
+    description: "The value of the MongoDB Key (see for details: https://docs.mongodb.org/v2.4/tutorial/generate-key-file/)"
+    generate: expression
+    from: "[a-zA-Z0-9]{255}"
+    required: true
+
+  - name: MONGODB_IMAGE
+    displayName: "MongoDB Docker Image"
+    description: "The name of the MongoDB docker image"
+    value: "openshift/mongodb-24-centos7"
+    required: true
+
+  - name: MONGODB_SERVICE_NAME
+    displayName: "OpenShift Service Name"
+    description: "The name of the MongoDB service, used for DNS lookup"
+    value: mongodb
+    required: true
+
+  - name: VOLUME_CAPACITY
+    displayName: "Volume Capacity"
+    description: "Volume space available for data, e.g. 512Mi, 2Gi"
+    value: "1Gi"
+    required: true
+
+objects:
+  # A headless service to create DNS records
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      name: "${MONGODB_SERVICE_NAME}"
+    spec:
+      # for a headless service proxying is not required
+      clusterIP: None
+      # the list of ports that are exposed by this service
+      ports:
+        - name: mongodb
+          port: 27017
+      # will route traffic to pods having labels matching this selector
+      selector:
+        node-type: "${MONGODB_SERVICE_NAME}-node"
```

Comment:
"node" has a special meaning in OpenShift, I suggest not using this word as part of the label.

Suggestion:

```yaml
"name": "${DATABASE_SERVICE_NAME}"
```

(same as https://github.com/openshift/origin/blob/master/examples/db-templates/mongodb-persistent-template.json)

@php-coder if you're not opposed I can change it.